### PR TITLE
GEOMESA-748 Attribute histogram analytic job

### PR DESCRIPTION
**Pull request by elahrvivaz**
* Converted old jobs to scalding type-safe API
* Fixed Accumulo and GeoMesa scalding sources to work with type-safe API
* Added a job that can copy a schema from one table/instance to another
* Added optional transforms to GeoMesaInputFormat

### CODE REVIEW

File: geomesa-jobs/README.md [hunk 1]
```diff
@@ -65,6 +65,43 @@ geomesa> mvn clean install -P assemble -pl geomesa-jobs
...
 argument to ensure the correct jars are available on the distributed classpath.
 
+#### Analytic Histograms
+
+GeoMesa provides the ability to analyze your data through histograms.
+
+The job can be invoked through yarn as follows (jar version may vary slightly):
+
+```shell
+geomesa> yarn jar geomesa-jobs/target/geomesa-jobs-accumulo1.5-1.0.0-shaded.jar \
```
Comment (anthonyccri) [inline geomesa-jobs/README.md]: There's quite a lot of functionality that we could expose if we supported Hive - it has ```SELECT AS``` and ```GROUPBY``` and ```COUNT DISTINCT```.  Thoughts on looking into supporting Hive?

Comment (elahrvivaz) [inline geomesa-jobs/README.md]: That sounds like exactly what @BrittEvans wants.

Comment (BrittEvans) [inline geomesa-jobs/README.md]: Yeah, pretty much.  Sounds interesting.

File: geomesa-tools/src/main/scala/org/locationtech/geomesa/tools/ingest/ScaldingDelimitedIngestJob.scala [hunk 2]
```diff
@@ -29,7 +29,7 @@ import org.geotools.factory.Hints
 import org.geotools.filter.identity.FeatureIdImpl
 import org.geotools.geometry.jts.JTSFactoryFinder
 import org.geotools.util.Converters
-import org.joda.time.DateTime
+import org.joda.time.{DateTimeZone, DateTime}
```
Comment (jdk2pq) [inline geomesa-tools/src/main/scala/org/locationtech/geomesa/tools/ingest/ScaldingDelimitedIngestJob.scala]: [picky] organize imports

File: geomesa-tools/src/test/scala/org/locationtech/geomesa/tools/ScaldingDelimitedIngestJobTest.scala [hunk 3]
```diff
@@ -18,7 +18,7 @@ package org.locationtech.geomesa.tools
 
 import java.util.{Date, UUID}
 
-import com.twitter.scalding.Args
+import com.twitter.scalding.{Mode, Args}
```
Comment (jdk2pq) [inline geomesa-tools/src/test/scala/org/locationtech/geomesa/tools/ScaldingDelimitedIngestJobTest.scala]: [picky] organize imports

**Comment by jdk2pq** [comment]
Looks awesome. :+1:

File: geomesa-tools/src/main/scala/org/locationtech/geomesa/tools/ingest/ScaldingDelimitedIngestJob.scala (hunk 2 above)
Comment (elahrvivaz) [inline geomesa-tools/src/main/scala/org/locationtech/geomesa/tools/ingest/ScaldingDelimitedIngestJob.scala]: fixed.

File: geomesa-tools/src/test/scala/org/locationtech/geomesa/tools/ScaldingDelimitedIngestJobTest.scala (hunk 3 above)
Comment (elahrvivaz) [inline geomesa-tools/src/test/scala/org/locationtech/geomesa/tools/ScaldingDelimitedIngestJobTest.scala]: fixed.

File: geomesa-jobs/src/main/scala/org/locationtech/geomesa/jobs/analytics/HistogramJob.scala [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+class HistogramJob(args: Args) extends GeoMesaBaseJob(args) {
+
+  val feature  = args(FEATURE_IN)
+  val dsParams = toDataStoreInParams(args)
+  val filter   = args.optional(CQL_IN)
+
+  val attribute  = args(ATTRIBUTE)
+  val groupBy    = args.nonStrictList(GROUP_BY)
+  val uniqueBy   = args.nonStrictList(UNIQUE_BY)
+  val transforms = Option(args.list(TRANSFORM_IN).toArray).filter(!_.isEmpty)
```
Comment (jdk2pq) [inline geomesa-jobs/src/main/scala/org/locationtech/geomesa/jobs/analytics/HistogramJob.scala]: `(_.nonEmpty)`

Comment (elahrvivaz) [inline geomesa-jobs/src/main/scala/org/locationtech/geomesa/jobs/analytics/HistogramJob.scala]: fixed.

File: geomesa-jobs/src/main/scala/org/locationtech/geomesa/jobs/analytics/HistogramJob.scala [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+      Class.forName("scala.Tuple" + l.size).getConstructors.apply(0).newInstance(l: _*).asInstanceOf[Product]
+  }
+
+  implicit class RichAttribute(val sf: SimpleFeature) extends AnyVal {
+
+    /**
+     * Gets a valid string for an attribute - 'null' as a string if the attribute is null
+     */
+    def safeString(attribute: String): String =
+      Option(sf.getAttribute(attribute)).map(_.toString).filter(!_.isEmpty).getOrElse("null")
```
Comment (jdk2pq) [inline geomesa-jobs/src/main/scala/org/locationtech/geomesa/jobs/analytics/HistogramJob.scala]: `nonEmpty`

File: geomesa-core/src/main/scala/org/locationtech/geomesa/core/index/index.scala [hunk 6]
```diff
@@ -80,6 +91,90 @@ package object index {
...
+
+  /**
+   * Checks for attribute transforms in the query and sets them as hints if found
+   *
+   * @param query
+   * @param sft
+   * @return
+   */
+  def setQueryTransforms(query: Query, sft: SimpleFeatureType) =
+    if (query.getProperties != null && !query.getProperties.isEmpty) {
```
Comment (jdk2pq) [inline geomesa-core/src/main/scala/org/locationtech/geomesa/core/index/index.scala]: `nonEmpty`

File: geomesa-core/src/main/scala/org/locationtech/geomesa/core/index/index.scala [hunk 7]
```diff
@@ -80,6 +91,90 @@ package object index {
...
+            ab.buildDescriptor(name, ab.buildType())
+          }
+      }
+    }
+
+    val geomAttributes = attributes.filter(_.isInstanceOf[GeometryAttribute]).map(_.getLocalName)
+    val sftBuilder = new SimpleFeatureTypeBuilder()
+    sftBuilder.setName(origSFT.getName)
+    sftBuilder.addAll(attributes.toArray)
+    if (!geomAttributes.isEmpty) {
```
Comment (jdk2pq) [inline geomesa-core/src/main/scala/org/locationtech/geomesa/core/index/index.scala]: `nonEmpty`

File: geomesa-jobs/src/main/scala/org/locationtech/geomesa/jobs/analytics/HistogramJob.scala (hunk 5 above)
Comment (elahrvivaz) [inline geomesa-jobs/src/main/scala/org/locationtech/geomesa/jobs/analytics/HistogramJob.scala]: nonEmpty on a string causes an extra implicit object to be created...

Comment (jdk2pq) [inline geomesa-jobs/src/main/scala/org/locationtech/geomesa/jobs/analytics/HistogramJob.scala]: :+1:  my bad.

File: geomesa-core/src/main/scala/org/locationtech/geomesa/core/index/index.scala (hunk 7 above)
Comment (elahrvivaz) [inline geomesa-core/src/main/scala/org/locationtech/geomesa/core/index/index.scala]: fixed.

File: geomesa-core/src/main/scala/org/locationtech/geomesa/core/index/index.scala (hunk 6 above)
Comment (elahrvivaz) [inline geomesa-core/src/main/scala/org/locationtech/geomesa/core/index/index.scala]: not a scala collection, requires a conversion...

**Comment by jdk2pq** [comment]
Committed as abd7f9e61a81891af84b23c81db624a343c66b95

**Pull request closed without merging**